File: benchmarks/fibonacci/fibonacci.cpp
#include <iostream>

constexpr auto fibonacci(u_int32_t n) {
  if (n <= 1) {
    return n;
  }
  u_int32_t prev = 0, current = 1;
  for (size_t i = 2; i <= n; i++) {
    u_int32_t next = prev + current;
    prev = current;
    current = next;
  }
  return current;
}

int main() {
  std::printf("%d", fibonacci(46));
}

File: benchmarks/fibonacci/fibonacci.rs
fn fibonacci(n: u32) -> u32 {
  if n <= 1 {
    return n;
  }
  let (mut prev, mut current) = (0, 1);
  for _ in 2..=n {
    let next = prev + current;
    prev = current;
    current = next;
  }
  current
}

fn main() {
  println!("{}", fibonacci(46));
}
